src/main/app/src/Components/Landscape/Overview/OverviewLayout.tsx

```diff
@@ -40,86 +40,93 @@ const useStyles = makeStyles((theme: Theme) =>
 interface Props {
   landscapes: ILandscape[] | null | undefined;
   setSidebarContent: Function;
+  landscapesCount: Number | null | undefined;
 }
 
 /**
  * Displays all available landscapes and provides all needed navigation
  */
 
-const OverviewLayout: React.FC<Props> = ({ landscapes, setSidebarContent }) => {
+const OverviewLayout: React.FC<Props> = ({ landscapes, setSidebarContent, landscapesCount }) => {
   const classes = useStyles();
   const componentClasses = componentStyles();
+  const history = useHistory();
   let content: ReactElement[] = [<Box>Loading landscapes...</Box>];
 
-  if (Array.isArray(landscapes) && landscapes.length) {
-    content = landscapes.map((landscape) => {
-      let itemCount = 0;
-      landscape.groups?.forEach((group) => (itemCount += group.items.length));
-      let stats =
-        itemCount +
-        ' items in ' +
-        (landscape.groups ? Object.keys(landscape.groups).length : 0) +
-        ' groups';
+  if (Array.isArray(landscapes) && landscapes.length && landscapesCount) {
+    if (landscapesCount > 1) {
```

Comment:
If needed at all, I would revert the if statement so that the "else" is not necessary:

if (count ==1) {
history....
return null;
}

Reply:
yes, I tried this but the problem here is the call to OveriewLayout is synchronous, which means whenever a single landscape is loaded OveriewLayout is getting called .we are not waiting to fetch all landscape and to display at once. so it is always satisfying if condition and "history.push" happens every time.